index.bs

```diff
@@ -308,15 +294,103 @@ The {{Blob()}} constructor can be invoked with the parameters below:
-  <dd>which takes one member:
+  <dd>which takes two members:
     * <dfn id="dfn-BPtype" dict-member for="BlobPropertyBag">type</dfn>,
       the ASCII-encoded string in lower case representing the media type of the {{Blob}}.
       Normative conditions for this member are provided in the [[#constructorBlob]].
+    * <dfn dict-member for="BlobPropertyBag">endings</dfn>,
+      an enum which can take the values {{"transparent"}} or {{"native"}}.
+      By default this is set to {{"transparent"}}. If set to {{"native"}},
+      [=convert line endings to native|line endings will be converted to native=]
+      in any {{USVString}} elements in {{blobParts}}.
 </dl>
 
+<div algorithm="process-blob-parts">
+To <dfn lt="process blob parts|processing blob parts">process blob parts</dfn> given a sequence of {{BlobPart}}'s |parts|
+and {{BlobPropertyBag}} |options|,
+run the following steps:
+
+1. Let |bytes| be an empty sequence of bytes.
+
+1. For each |element| in |parts|:
+
+  1. If |element| is a {{USVString}}, run the following substeps:
+
+    1. Let |s| be |element|.
+
+    1. If the {{BlobPropertyBag/endings}} member of |options| is set to {{"native"}},
+      set |s| to the result of [=converting line endings to native=] of |element|.
+
+    1. Append the result of [=UTF-8 encoding=] |s| to |bytes|.
+
+      Note: The algorithm from WebIDL [[WebIDL]] replaces unmatched surrogates in an invalid utf-16 string
+      with U+FFFD replacement characters.
+      Scenarios exist when the {{Blob}} constructor may result in some data loss
+      due to lost or scrambled character sequences.
+
+  1. If |element| is a {{BufferSource}}, <a lt="get a copy of the buffer source">get
+    a copy of the bytes held by the buffer source</a>, and append those bytes to |bytes|.
+
+  1. If |element| is a {{Blob}},
+    append the bytes it represents to |bytes|.
+
+    Note: The {{Blob/type}} of the {{Blob}} array element is ignored and will not affect {{Blob/type}} of returned
+    {{Blob}} object.
+
+1. Return |bytes|.
+
+</div>
+
+<div algorithm="convert-line-endings-to-native">
+To <dfn lt="convert line endings to native|converting line endings to native">
+convert line endings to native</dfn> in a [=string=] |s|,
+run the following steps:
+
+1. Let |native line ending| be be the [=code point=] U+000A LF.
+
+1. If it matches the underlying platform's conventions,
+  set |native line ending| to the [=code point=] U+000D CR
+  followed by the [=code point=] U+000A.
+
+1. Set |result| to the empty [=string=].
+
+1. Let |position| be a [=position variable=] for |s|,
+  initially pointing at the start of |s|.
+
+1. Let |token| be the result of [=collecting a sequence of code points=]
+  that are not equal to U+000A LF or U+000D CR
+  from |s| given |position|.
+
+1. Append |token| to |result|.
+
+1. While |position| is not past the end of |s|:
+
+  1. If the [=code point=] at |position| within |s| equals U+000D CR:
```

Comment:
Subsequent steps would be simpler if you assign a local variable (e.g. _cp_) to _code point at position within s_

Reply:
Not sure about that. It's only used twice (at least with the Otherwise added below, else it would only be referenced once) so I don't think we gain much by adding a temporary local variable.